src/Host/Broker/Impl/Sessions/Session.cs

```diff
@@ -69,41 +69,49 @@ public class Session {
             CommandLineArguments = commandLineArguments;
         }
 
-        public void StartHost(SecureString password, ILogger outputLogger, ILogger messageLogger) {
+        public void StartHost(SecureString password, string profilePath, ILogger outputLogger, ILogger messageLogger) {
             string brokerPath = Path.GetDirectoryName(typeof(Program).Assembly.GetAssemblyPath());
             string rhostExePath = Path.Combine(brokerPath, RHostExe);
+            string arguments = $"--rhost-name \"{Id}\" {CommandLineArguments}";
+            var username = new StringBuilder(NativeMethods.CREDUI_MAX_USERNAME_LENGTH + 1);
+            var domain = new StringBuilder(NativeMethods.CREDUI_MAX_PASSWORD_LENGTH + 1);
 
-            var psi = new ProcessStartInfo(rhostExePath) {
+            if (NativeMethods.CredUIParseUserName(User.Name, username, username.Capacity, domain, domain.Capacity) != 0) {
+                throw new ArgumentException();
+            }
+
+            ProcessStartInfo psi = new ProcessStartInfo(rhostExePath) {
                 UseShellExecute = false,
                 CreateNoWindow = false,
-                Arguments = $"--rhost-name \"{Id}\" {CommandLineArguments}",
+                Arguments = arguments,
                 RedirectStandardInput = true,
                 RedirectStandardOutput = true,
-                RedirectStandardError = true
+                RedirectStandardError = true,
+                LoadUserProfile = true
             };
 
+            if (WindowsIdentity.GetCurrent().User != ((WindowsIdentity)User).User) {
```

Comment:
Does `((WindowsIdentity)User)` cast work in local scenarios when `--security:secret` is used? i.e. is it still a `WindowsIdentity` then?